scuba/config.py

```diff
@@ -65,7 +91,31 @@ def from_yaml(self, node):
             raise yaml.YAMLError('Key "{}" not found in {}'.format(key, filename))
         return cur
 
+    def override(self, node):
+        '''
+        Implements !override constructor
+        '''
+        # Load the content from the node, as a scalar
+        content = self.construct_scalar(node)
+
+        # Dynamically add an OverrideMixin to the resulting object's type
+        obj = yaml.load(content, lambda s: Loader(s, root=self._root))
+        if obj is None:
+            obj = OverrideNone()
+        else:
+            objtype = type(obj)
+            mixin_type = type('Override' + objtype.__name__, (objtype, OverrideMixin), dict())
+
+            try:
+                obj.__class__ = mixin_type
+            except TypeError:
+                # Primitive classes (e.g., int, str) don't support __class__ assignment
+                obj = mixin_type(obj)
```

Comment:
:clap:  Very cool technique!

I've seen recommendations *against* changing the type of an existing object (by assigning to `__class__`). However, I think most of the trouble can come from the fact that you may be missing functions/attributes that should be there if you actually instantiated the class. So in this case, I don't really see any problem with it, because the `OverrideMixin` doesn't have any methods and doesn't expect or *do* anything.